Packages/Status/Sources/Status/Editor/StatusEditorViewModel.swift

```diff
@@ -532,93 +557,111 @@ import SwiftUI
   // MARK: - Media related function
 
   private func indexOf(container: StatusEditorMediaContainer) -> Int? {
-    mediasImages.firstIndex(where: { $0.id == container.id })
+    mediaContainers.firstIndex(where: { $0.id == container.id })
   }
 
-  func inflateSelectedMedias() {
-    mediasImages = []
+  func prepareToPost(for pickerItem: PhotosPickerItem) {
+    Task(priority: .high) {
+      if let container = await makeMediaContainer(from: pickerItem) {
+        await MainActor.run { self.mediaContainers.append(container) }
```

Comment:
Is it necessary? `StatusEditorViewModel` being `MainActor` should be enough for the property mutation to happen on the main thread.